lib/diagnostic.js

```diff
@@ -20,7 +31,18 @@ const sum = function sum() {
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
-
+  let newArr = [];
+  if(defaultValue instanceof Array) {
```

Comment:
I don't know that `defaultValue` is necessarily only an array or a function - it could easily be a number or a string.